File: 프로그래머스/2/84512. 모음 사전/모음 사전.java
import java.util.*;
class Solution {
    ArrayList<String> list;
    String[]  words = {"A","E","I","O","U"};
    public int solution(String word) {
        list= new ArrayList<>();
       
        
        dfs("",0);
        int answer = list.indexOf(word);
        return answer;
    }
    void dfs(String word, int depth) {
        list.add(word);
        if(depth == 5) return;
        for(int i = 0; i < 5; i++) {
            dfs(word+words[i],depth+1);
        }
    }
}

File: 프로그래머스/2/84512. 모음 사전/모음 사전.js
function solution(word) {
    var answer = 0;
    const map = [];
    const words = ['A','E','I','O','U'];
    function dfs(w,depth) {
        map.push(w);
        if(depth === words.length) {
            return;
        }
        for (let i = 0; i < words.length; i++) {
            dfs(w + words[i],depth + 1);
        }
    }
    dfs('',0);
    answer = map.indexOf(word);
    return answer;
}